crates/rethnet_rpc_server/src/lib.rs

```diff
@@ -560,17 +594,46 @@ async fn router(state: StateType) -> Router {
 
 impl Server {
     /// accepts a configuration and a set of initial accounts to initialize the state.
-    pub async fn new(
-        config: Config,
-        genesis_accounts: HashMap<Address, AccountInfo>,
-    ) -> Result<Self, Error> {
+    pub async fn new(config: Config) -> Result<Self, Error> {
         let listener = TcpListener::bind(config.address).map_err(Error::Listen)?;
         event!(Level::INFO, "Listening on {}", config.address);
 
-        let accounts: Vec<Address> = genesis_accounts
-            .iter()
-            .map(|(address, _account_info)| *address)
-            .collect();
+        let (local_accounts, genesis_accounts): (
+            HashMap<Address, SecretKey>,
+            HashMap<Address, AccountInfo>,
+        ) = {
+            let secp256k1 = Secp256k1::signing_only();
+            config.accounts.iter().enumerate().fold(
```

Comment:
Instead of using `fold`, which can result in allocations per insertion, this would be more efficient by using `map` and `collect`ing a pair of pairs: `((Address, U256), (Address, AccountInfo))`

Reply:
addressed in https://github.com/NomicFoundation/hardhat/pull/4149/commits/ac0a31b14125d831ab243a8c4d3921c383f8147b